grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServiceServerTest.java

```diff
@@ -936,10 +936,11 @@ public void onCompleted() {
 
         final RpcResponse rpcResponse = (RpcResponse) log.responseContent();
         final StatusRuntimeException cause = (StatusRuntimeException) rpcResponse.cause();
-        assertThat(cause.getStatus().getCode()).isEqualTo(Code.CANCELLED);
         if (protocol.isMultiplex()) {
+            assertThat(cause.getStatus().getCode()).isEqualTo(Code.CANCELLED);
             assertThat(cause.getStatus().getCause()).isInstanceOf(ClosedStreamException.class);
         } else {
+            assertThat(cause.getStatus().getCode()).isEqualTo(Code.UNKNOWN);
```

Comment:
I think this is the limitation of gRPC with HTTP/1.1.
We shouldn't treat a closed connection as a cancellation from the client.
Please let me know if you have a different thought.

Reply:
I think this can go either way.

I'm slightly in favor of the previous behavior since:
- Looking at the API names, I think it's reasonable to assume that calling `ClientRequestContext#cancel` will result in a `CANCELLED` status
- `CANCELLED` doesn't necessarily mean "cancellation from the client.". For instance, a timeout due to `RequestTimeoutException` is also considered `CANCELLED`. Network failure in the upstream implementation is also considered `CANCELLED`.

Having said this, I'm not too strong on this.

Reply:
>  Network failure in the upstream implementation is also considered CANCELLED.

Could you give me the link to it? If that's true, I think we can change to return canceled when a ClosedSesessionException is raised here:
https://github.com/line/armeria/blob/1e1439664309f457458517c42dd38e201e98cad7/grpc/src/main/java/com/linecorp/armeria/common/grpc/DefaultGrpcExceptionHandlerFunction.java#L53-L58

Reply:
Actually, I can't say I'm too confident of my previous statement.

It is based on the seeing the following lines
https://github.com/grpc/grpc-java/blob/db96219be1ed6a996ed3a4cd8812446166118c06/stub/src/main/java/io/grpc/stub/ServerCalls.java#L287-L292
and the following issue:
https://github.com/grpc/grpc-java/issues/9682#issuecomment-1309254805

Having said this, I haven't actually tested this out.

Reply:
I see. Then let's wait for @ikhoon who implemented the logic.

Reply:
Between `UNKNOWN` and `CANCELLED`, I slightly prefer `CANCELLED` to synchronize with the upstream behavior.

Reply:
Have checked that the upstream uses `CANCELLED` when a connection is closed.
Thus, use the same approach.